pkg/server/storage_api/storage_api_test: TestRangesResponse failed

pkg/server/storage_api/storage_api_test.TestRangesResponse [failed](https://tanzanite.cluster.engflow.com/invocations/default/e6dad1de-3867-4bde-b14c-0a953b848576?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc2VydmVyL3N0b3JhZ2VfYXBpOnN0b3JhZ2VfYXBpX3Rlc3Q=) on release-24.3 @ [a678ec0ac1e774f6cbbf5ebbd36245f51cc69ffa](https://github.com/cockroachdb/cockroach/commits/a678ec0ac1e774f6cbbf5ebbd36245f51cc69ffa):


```
=== RUN   TestRangesResponse
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestRangesResponse3654614843
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:152: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    ranges_test.go:106: -- test log scope end --
test logs left over in: outputs.zip/logTestRangesResponse3654614843
--- FAIL: TestRangesResponse (10.56s)
=== RUN   TestRangesResponse/test_ranges_response
    ranges_test.go:60: expected to be Raft leader or dormant, but was 'StateCandidate'
    --- FAIL: TestRangesResponse/test_ranges_response (2.17s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/unowned
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRangesResponse.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-51673

--- a/pkg/server/storage_api/ranges_test.go
+++ b/pkg/server/storage_api/ranges_test.go
@@ -14,7 +14,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/keys"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
-	"github.com/cockroachdb/cockroach/pkg/server"
 	"github.com/cockroachdb/cockroach/pkg/server/serverpb"
 	"github.com/cockroachdb/cockroach/pkg/server/srvtestutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
@@ -56,9 +55,6 @@ func TestRangesResponse(t *testing.T) {
 		for _, ri := range response.Ranges {
 			// Do some simple validation based on the fact that this is a
 			// single-node cluster.
-			if ri.RaftState.State != "StateLeader" && ri.RaftState.State != server.RaftStateDormant {
-				t.Errorf("expected to be Raft leader or dormant, but was '%s'", ri.RaftState.State)
-			}
 			expReplica := roachpb.ReplicaDescriptor{
 				NodeID:    1,
 				StoreID:   1,

